test/s3path.jl

```diff
@@ -354,4 +354,27 @@ end
     rm(json_path)
 end
 
+# <https://github.com/JuliaCloud/AWSS3.jl/issues/168>
+@testset "Global config is not frozen at construction time" begin
+    # Setup: create a file holding a string `abc`
+    path = S3Path("s3://$(bucket_name)/test_str.txt")
+    write(path, "abc")
+
+    prev_config = global_aws_config()
+    try
+        global_aws_config(region="us-east-2") # this is the wrong region!
+        path = S3Path("s3://$(bucket_name)/test_str.txt")
+        @test_throws AWS.AWSException read(path, String)
```

Comment:
Test is frail and will not throw an exception if the default region is set to "us-east-2".
```suggestion
    prev_config = global_aws_config()
    
    # Setup: create a file holding a string `abc`
    path = S3Path("s3://$(bucket_name)/test_str.txt")
    write(path, "abc")
    @test read(path, String) == "abc"  # Have access to read file

    alt_region = prev_config.region == "us-east-2" ? "us-east-1" : "us-east-2"
    try
        global_aws_config(region=alt_region) # this is the wrong region!
        @test_throws AWS.AWSException read(path, String)
```

Reply:
good point!